protocol/contracts/amo/TpfAmo.sol

```diff
@@ -0,0 +1,549 @@
+                1, minAmountOut, templeBalancerPoolIndex == 0 ? 1 : 0);
+        } else {
+            exitPoolRequest = poolHelper.createPoolExitRequest(address(temple), address(stable), bptAmountIn,
+                0, minAmountOut, templeBalancerPoolIndex == 0 ? 1 : 0);
+        }
+
+        // withdraw and unwrap deposit token to BPT token
+        amoStaking.withdrawAndUnwrap(bptAmountIn, false, true);
+
+        // safeincrease allowance
+        bptToken.approve(address(balancerVault), 0);
+        bptToken.approve(address(balancerVault), bptAmountIn);
+
+        // execute call and check for sanity
+        uint256 stableBalanceBefore = stable.balanceOf(address(this));
+        uint256 spotPriceScaledBefore = poolHelper.getSpotPriceScaled();
+        balancerVault.exitPool(balancerPoolId, address(this), address(this), exitPoolRequest);
+        uint256 stableBalanceAfter = stable.balanceOf(address(this));
+        if (stableBalanceAfter < stableBalanceBefore + minAmountOut) {
+            revert AMOCommon.InsufficientAmountOutPostcall(stableBalanceBefore + minAmountOut, stableBalanceAfter);
+        }
+
+        // revert if high slippage after pool join
+        if (uint64(poolHelper.getSlippage(spotPriceScaledBefore)) > postRebalanceSlippage) {
+            revert AMOCommon.HighSlippage();
+        }
+        lastRebalanceTimeSecs = uint64(block.timestamp);
+        emit WithdrawStable(bptAmountIn, minAmountOut);
+    }
+
+    /**
+     * @notice Add liquidity with both TEMPLE and stable tokens into balancer pool. 
+     * Treasury Price Floor is expected to be within bounds of multisig set range.
+     * BPT tokens are then deposited and staked in Aura.
+     * @param request Request data for joining balancer pool. Assumes userdata of request is
+     * encoded with EXACT_TOKENS_IN_FOR_BPT_OUT type
+     * @param minBptOut Minimum amount of BPT tokens expected to receive
+     */
+    function addLiquidity(
+        AMO__IBalancerVault.JoinPoolRequest memory request,
+        uint256 minBptOut
+    ) external onlyOwner {
+        // validate request
+        if (request.assets.length != request.maxAmountsIn.length || 
+            request.assets.length != 2 || 
+            request.fromInternalBalance == true) {
+                revert AMOCommon.InvalidBalancerVaultRequest();
+        }
+
+        uint256 templeAmount;
+        uint256 stableAmount;
+        if (templeBalancerPoolIndex == 0) {
+            templeAmount = request.maxAmountsIn[0];
+            stableAmount = request.maxAmountsIn[1];
+        } else {
+            templeAmount = request.maxAmountsIn[1];
+            stableAmount = request.maxAmountsIn[0];
+        }
+
+        AMO__ITempleERC20Token(address(temple)).mint(address(this), templeAmount);
+        // safe allowance stable and TEMPLE
+        temple.safeIncreaseAllowance(address(balancerVault), templeAmount);
```

Comment:
Same as other comment - why temple uses safeIncreaseAllowance, but stable is a fixed approve?